ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/AggregateAnalyzer.java

```diff
@@ -359,5 +401,14 @@ public Void visitQualifiedColumnReference(
     ) {
       throw new UnsupportedOperationException("Should of been converted to unqualified");
     }
+
+    @Override
+    public Void visitSubscriptExpression(final SubscriptExpression node, final Void context) {
```

Comment:
are there any other expressions that we should be looking that can match exactly other than function calls and subscript expressions? i.e. what about arithmetic expressions (`SELECT AS_VALUE(a + b) AS foo GROUP BY a + b`)

I think it makes sense to implement this in `visitExpression` instead of in each individual expression type. We would only "override" this behavior when we need to (e.g. inside `visitFuncitonCall`)

Reply:
Yes, this is a good point in the sense that we need to figure out which expressions need special handling, I will try the arithmetic expressions. However, there is no `visitExpression` in `TraversalExpressionVisitor` unless I misunderstand what you meant.

Reply:
@agavra The approach of putting the "common" code in `process` doesn't work. Reason being is that if the parent expression exists in the group by e.g. `(a+b)`, it will set the flag `foundExpressionInGroupBy` to true and will continue visiting the children, `a` and `b`. If a child doesn't exist in the group by, it will set the flag to false. All the rest of the children then (`b` in this example), will fail and will wrongly get added to the `nonAggParams`. 
TLDR: the flag must be set to false only after the entire tree of the expression has been visited. And that is only possible to do in the specific method for each expression

Reply:
look at my newest comment, I think we have that problem anyway. I still think we should be able to do this in the common process if you take the suggestion I have there.

Reply:
https://github.com/confluentinc/ksql/pull/5998#discussion_r473371482